runtime/lua/vim/diagnostic.lua

```diff
@@ -478,6 +478,10 @@ local function set_list(loclist, opts)
   -- numbers beyond the end of the buffer
   local diagnostics = get_diagnostics(bufnr, opts, false)
   local items = M.toqflist(diagnostics)
+  if #items == 0 then
```

Comment:
`#` works for serialized table . so use `next(item) == nil ` or `vim.tbl_count(item) == 0 `

Reply:
Done!

Reply:
I think you'd better add a test for this. ref

https://github.com/neovim/neovim/blob/a6b05cb75d330dd995d3ad21ee08bb0a2cfcae74/test/functional/lua/diagnostic_spec.lua#L1975

Reply:
But I have no idea how to add tests, because new implementation does't produce any data and return directly.😂 Could you give me some suggestion?

Reply:
just take a look at the test case, there is no test case for `vim.diagnostic.setqflist`  🥲  maybe no need . wait the @gpanders review

Reply:
> `#` works for serialized table . so use `next(item) == nil ` or `vim.tbl_count(item) == 0`

This table is a list like table, so using `#` is fine (and preferred, as its intention is clearer than `next()`).